src/Draco.Compiler/Internal/OptimizingIr/Model/Assembly.cs

```diff
@@ -38,37 +32,13 @@ internal sealed class Assembly : IAssembly
     }
     IProcedure? IAssembly.EntryPoint => this.EntryPoint;
 
-    private readonly Dictionary<GlobalSymbol, Global> globals = new();
-    private readonly Dictionary<FunctionSymbol, IProcedure> procedures = new();
-    private Procedure? entryPoint;
+    public IModule RootModule => this.rootModule;
```

Comment:
Why not follow the pattern as for others, you can explicitly implement this interface member, then have a public mutable one, so the mutable assembly can always give you a mutable root, avoiding a few casts:
```suggestion
    public Module RootModule => this.rootModule;
    IModule IAssembly.RootModule => this.RootModule;
```